packages/actor-http-retry/lib/ActorHttpRetry.ts

```diff
@@ -0,0 +1,178 @@
+import type { IActionHttp, IActorHttpOutput, IActorHttpArgs, MediatorHttp } from '@comunica/bus-http';
+import { ActorHttp } from '@comunica/bus-http';
+import { KeysHttp } from '@comunica/context-entries';
+import { ActionContextKey, passTest, failTest } from '@comunica/core';
+import type { TestResult } from '@comunica/core';
+import type { IMediatorTypeTime } from '@comunica/mediatortype-time';
+
+export class ActorHttpRetry extends ActorHttp {
+  private readonly mediatorHttp: MediatorHttp;
+  private readonly activeDelays: Record<string, { date: Date; timeout: NodeJS.Timeout }>;
+
+  // Expression that matches dates expressed in the HTTP Date header format
+  // eslint-disable-next-line max-len
+  private static readonly dateRegex = /^(Mon|Tue|Wed|Thu|Fri|Sat|Sun), [0-9]{2} (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) [0-9]{4} [0-9]{2}:[0-9]{2}:[0-9]{2} GMT$/u;
+  private static readonly numberRegex = /^[0-9]+$/u;
+  private static readonly keyWrapped = new ActionContextKey<boolean>('urn:comunica:actor-http-retry#wrapped');
+
+  public constructor(args: IActorHttpQueueArgs) {
+    super(args);
+    this.mediatorHttp = args.mediatorHttp;
+    this.activeDelays = {};
```

Comment:
We'll need a http cache invalidation observer on here, to reset all delays if we forcefully invalidate comunica's cache.

Reply:
There is now an invalidation observer, which is also being tested by the unit tests.